tapdb/multiverse.go

```diff
@@ -443,6 +528,44 @@ func (b *MultiverseStore) queryRootNodes(ctx context.Context,
 	return uniRoots, nil
 }
 
+// fillSyncerCache populates the syncer cache with all the root nodes that are
+// currently known. This is used to quickly serve the syncer with the root nodes
+// it needs to sync the multiverse.
+//
+// NOTE: This method must be called while holding the syncer cache lock.
+func (b *MultiverseStore) fillSyncerCache(ctx context.Context) error {
+	now := time.Now()
+	log.Infof("Populating syncer root cache...")
+
+	params := sqlc.UniverseRootsParams{
+		SortDirection: sqlInt16(universe.SortAscending),
+		NumOffset:     0,
+		NumLimit:      universe.MaxPageSize,
```

Comment:
In cases like this, I wonder if it's better to just read the entire set of roots in one go. So one single query, instead of a bunch of smaller ones. With the current amount of assets on mainnet, we'll end up doing 300 or so queries here to read them all out. Some benchmarking would likely help us find the sweet spot here re page size to read. 

Non-blocking.